src/metabase/api/embed.clj

```diff
@@ -372,3 +374,42 @@
                                                          :api (api.embed.common/parse-query-params query-params)
                                                          :qp qp.pivot/run-pivot-query)
     (events/publish-event! :event/card-read {:object-id card-id, :user-id api/*current-user-id*, :context :dashboard})))
+
+(api.macros/defendpoint :get "/tiles/card/:token/:zoom/:x/:y/:lat-field/:lon-field"
+  "Generates a single tile image for an embedded Card using the map visualization."
+  [{:keys [token zoom x y lat-field lon-field]}
+   :- [:map
+       [:token     string?]
+       [:zoom      ms/Int]
+       [:x         ms/Int]
+       [:y         ms/Int]
+       [:lat-field ::api.tiles/field-id-or-name]
+       [:lon-field ::api.tiles/field-id-or-name]]
+   {:keys [parameters]}
+   :- [:map
+       [:parameters {:optional true} ms/JSONString]]]
+  (let [unsigned   (unsign-and-translate-ids token)
+        card-id    (embed/get-in-unsigned-token-or-throw unsigned [:resource :question])
+        parameters (json/decode+kw parameters)]
+    (api.embed.common/check-embedding-enabled-for-card card-id)
+    (api.tiles/process-tiles-query-for-card card-id parameters zoom x y lat-field lon-field)))
+
+(api.macros/defendpoint :get "/tiles/dashboard/:token/dashcard/:dashcard-id/card/:card-id/:zoom/:x/:y/:lat-field/:lon-field"
+  "Generates a single tile image for a Card on an embedded Dashboard using the map visualization."
+  [{:keys [token dashcard-id card-id zoom x y lat-field lon-field]}
+   :- [:map
+       [:token       string?]
+       [:dashcard-id ms/PositiveInt]
+       [:card-id     ms/PositiveInt]
+       [:zoom        ms/Int]
+       [:x           ms/Int]
+       [:y           ms/Int]
+       [:lat-field   ::api.tiles/field-id-or-name]
+       [:lon-field   ::api.tiles/field-id-or-name]]
```

Comment:
Just spitballing, but you could extract and register:

```
(mr/def :tiles/route-params
 [:map
  [:zoom        ms/Int]
  [:x           ms/Int]
  [:y           ms/Int]
  [:lat-field   ::api.tiles/field-id-or-name]
  [:lon-field   ::api.tiles/field-id-or-name]])
```

and then use merge in here:

```
   :- [:merge
       :tiles/route-params
       [:map
        [:token       string?]
        [:dashcard-id ms/PositiveInt]
        [:card-id     ms/PositiveInt]]
```